bundles/ch.elexis.core.services/src/ch/elexis/core/services/OrderService.java

```diff
@@ -49,15 +67,145 @@ public List<IOrderEntry> findOrderEntryForStock(IStock stock) {
 	public IOrderEntry addRefillForStockEntryToOrder(IStockEntry ise, IOrder order) {
 		int current = ise.getCurrentStock();
 		int max = ise.getMaximumStock();
-		if (max == 0) {
+		if (max == 0)
 			max = ise.getMinimumStock();
-		}
 		int toOrder = max - current;
 		if (toOrder > 0) {
-			IOrderEntry orderEntry = order.addEntry(ise.getArticle(), ise.getStock(), ise.getProvider(), toOrder);
-			modelService.save(orderEntry);
-			return orderEntry;
+			IOrderEntry entry = order.addEntry(ise.getArticle(), ise.getStock(), ise.getProvider(), toOrder);
+			modelService.save(entry);
+			getHistoryService().logChangedAmount(order, entry, 0, toOrder);
+			return entry;
 		}
 		return null;
 	}
-}
+
+	@Override
+	public Map<IArticle, Integer> calculateDailyConsumption(LocalDate date, List<IMandator> mandators) {
+		Map<IArticle, Integer> result = new LinkedHashMap<>();
+		IQuery<IEncounter> query = modelService.getQuery(IEncounter.class);
+		query.and(ModelPackage.Literals.IENCOUNTER__DATE, COMPARATOR.EQUALS, date);
+		if (mandators != null && !mandators.isEmpty()) {
+			query.startGroup();
+			for (IMandator m : mandators) {
+				query.or(ModelPackage.Literals.IENCOUNTER__MANDATOR, COMPARATOR.EQUALS, m);
+			}
+			query.andJoinGroups();
+		}
+		for (IEncounter encounter : query.execute()) {
+			for (IBilled billed : encounter.getBilled()) {
+				IBillable billable = billed.getBillable();
+				if (billable instanceof IArticle art) {
+					int amount = (int) billed.getAmount();
+					result.merge(art, amount, Integer::sum);
+				}
+			}
+		}
+		return result;
+	}
+
+	@Override
+	public void reduceOpenEntries(List<IOrder> orders, IArticle article, int reduceBy) {
+		for (IOrder order : orders) {
+			List<IOrderEntry> openEntries = order.getEntries().stream().filter(e -> e.getArticle().equals(article))
+					.filter(e -> e.getState() == OrderEntryState.OPEN).collect(Collectors.toList());
+			for (IOrderEntry entry : openEntries) {
+				if (reduceBy <= 0)
+					return;
+				int amount = entry.getAmount();
+				if (amount <= reduceBy) {
+					reduceBy -= amount;
+					getHistoryService().logRemove(order, entry);
+					modelService.delete(entry);
+				} else {
+					entry.setAmount(amount - reduceBy);
+					modelService.save(entry);
+					getHistoryService().logChangedAmount(order, entry, amount, entry.getAmount());
+					reduceBy = 0;
+				}
+			}
+		}
+	}
+
+	@Override
+	public void createOrderEntries(List<IOrder> existingOrders, IOrder fallbackOrder,
```

Comment:
Die Methode scheint die entries aller `existingOrders` nach offenen zu durchsuchen, und wenn ein passender Artikel in `entriesToAdd` gefunden wurde wird die Anzahl angepasst. Wenn kein passender gefunden wurde, wird ein neuer entry in den `fallbackOrder` erzeugt.

Anhand des Methoden Name hätte ich das nicht erwartet. Das müsste dann eher `addOrCreateOrderEntries` sein. Und dann gibts die `existingOrders` und den `createOrder` anstelle `fallbackOrder`.

Bitte Name und javadoc aktualisieren, damit leichter verständlich ist was hier geschieht.